Source/Concurrency/CivlRewriter.cs

```diff
@@ -43,9 +43,17 @@ public static void Transform(ConcurrencyOptions options, CivlTypeChecker civlTyp
       }
 
       // Desugaring of yielding procedures
-      YieldingProcChecker.AddCheckers(civlTypeChecker, decls);
-
-      if (!options.TrustSequentialization)
+      if (!options.TrustInvariants)
+      {
+         YieldingProcChecker.AddInvariantCheckers(civlTypeChecker, decls);
+      }
+      
+      if (!options.TrustRefinement)
+      {
+         YieldingProcChecker.AddRefinementCheckers(civlTypeChecker, decls);
+      }
+      
+      if (!options.TrustSequentialization && !options.TrustRefinement)
```

Comment:
Please move this code inside the then case of line 51 so that it is more obvious that one is a subcase of the other.